CMakeLists.txt

```diff
@@ -9,20 +9,16 @@ project(
 # ---- Warning guard ----
 
 # Protect dependents from this project's warnings if the guard isn't disabled
-set(example_warning_guard SYSTEM)
-if(example_INCLUDE_WITHOUT_SYSTEM)
-  set(example_warning_guard "")
+option(example_INCLUDE_WITHOUT_SYSTEM "Use SYSTEM keyword for include" OFF)
+if(NOT example_INCLUDE_WITHOUT_SYSTEM)
+  set(example_warning_guard SYSTEM)
 endif()
 
 # ---- Declare library ----
 
 include(GenerateExportHeader)
 
-set(headers include/example/main.h)
-
-set(sources source/main.cpp)
-
-add_library(example_example ${sources} ${headers})
+add_library(example_example source/main.cpp include/example/main.h)
```

Comment:
I would not recommend others to use variables

Reply:
Would be nice to have a rationale for this change. From my experience, CLion preferred to auto append sources to variables when creating new source files.

Reply:
Of course. The rationale for not using variables is the risk of typos being undiscovered. Assume you write the variable name wrong when passing it to the add_library call, CMake wont notice it, because it just evaluates to nothing. difficult to find, especially since the variables are case sensitive.

On the other hand, if you list the files explicitly in the add_library call, and have a typo there, CMake will detect this as an not existing file and throw an error.

Reply:
This is something that in my opinion comes down to the way people write CMLs. In my case, I use an editor that highlights all instances of a word once one is selected, so it's quite obvious when there is a typo and the auto completion takes all words into account, so it's easy to start typing a few letters and have the word you want be the first tab completion.

Missing sources also cause heaps of issues, so they will most definitely crop up if not in CMake.